### PR TITLE
split registration permissions

### PR DESCRIPTION
**Pull request by haoqing0110**
Signed-off-by: haoqing0110 <qhao@redhat.com>

### CODE REVIEW

File: manifests/klusterlet/managed/klusterlet-registration-clusterrole.yaml [hunk 1]
```diff
@@ -1,28 +1,16 @@
...
-  verbs: ["create"]
 # Allow agent to list clusterclaims
 - apiGroups: ["cluster.open-cluster-management.io"]
   resources: ["clusterclaims"]
-  verbs: ["get", "list", "watch"]
-# Allow agent to list addons lease
-- apiGroups: ["coordination.k8s.io"]
-  resources: ["leases"]
-  verbs: ["get", "list", "watch"]
+  verbs: ["get", "list", "watch"]
```
Comment (qiujian16) [inline manifests/klusterlet/managed/klusterlet-registration-clusterrole.yaml]: eof

File: manifests/klusterlet/managed/klusterlet-registration-clusterrole-addon-management.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+rules:
+# Allow agent to get/list/watch/create/delete/update/patch secrets.
+# TODO(zhujian7): consider to delete when addon agents can be deployed on the management cluster.
+- apiGroups: [""]
+  resources: ["secrets"]
+  verbs: ["get", "list", "watch", "create", "delete", "update", "patch"]
+# Allow agent to list addons lease
+- apiGroups: ["coordination.k8s.io"]
+  resources: ["leases"]
+  verbs: ["create", "get", "list", "update", "watch", "patch"]
```
Comment (qiujian16) [inline manifests/klusterlet/managed/klusterlet-registration-clusterrole-addon-management.yaml]: eof

File: manifests/klusterlet/managed/klusterlet-registration-clusterrolebinding-addon-management.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# ClusterRoleBinding for registration addon management permissions.
+# TODO: remove this ClusterRoleBinding, create RoleBindings and bind ClusterRole and ServiceAccount to user defined addon namespaces.
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
```
Comment (qiujian16) [inline manifests/klusterlet/managed/klusterlet-registration-clusterrolebinding-addon-management.yaml:7]: we should not need this when addon-management feature gate is not enabled.

**Review by qiujian16 — COMMENTED** [review]
I think we still need a secret management permission in namespace of the registration when addon-management is not needed. right?

**Comment by haoqing0110** [comment]
> I think we still need a secret management permission in namespace of the registration when addon-management is not needed. right?

yes, it's defined in https://github.com/open-cluster-management-io/registration-operator/pull/250/files#diff-b807821ba4bfaebd0ca64c3b0d36d3fd9430173f2b7feecd2c088fea7530f9e2R13

File: manifests/klusterlet/managed/klusterlet-registration-clusterrolebinding-addon-management.yaml (hunk 3 above)
Comment (haoqing0110) [reply manifests/klusterlet/managed/klusterlet-registration-clusterrolebinding-addon-management.yaml:7]: do you want me to add this to comments?

File: manifests/klusterlet/managed/klusterlet-registration-clusterrole.yaml (hunk 1 above)
Comment (haoqing0110) [reply manifests/klusterlet/managed/klusterlet-registration-clusterrole.yaml]: fixed

File: manifests/klusterlet/managed/klusterlet-registration-clusterrole-addon-management.yaml (hunk 2 above)
Comment (haoqing0110) [reply manifests/klusterlet/managed/klusterlet-registration-clusterrole-addon-management.yaml]: fixed

File: manifests/klusterlet/managed/klusterlet-registration-clusterrolebinding-addon-management.yaml (hunk 3 above)
Comment (haoqing0110) [reply manifests/klusterlet/managed/klusterlet-registration-clusterrolebinding-addon-management.yaml:7]: @qiujian16 add more comments for that.

**Pull request merged**